helix-lsp/src/lib.rs

```diff
@@ -320,56 +321,51 @@ impl Registry {
             .map(|(_, client)| client.as_ref())
     }
 
+    pub fn restart(&mut self, language_config: &LanguageConfiguration) -> Result<Arc<Client>> {
```

Comment:
The old LS will die when references disappear. https://github.com/helix-editor/helix/blob/master/helix-lsp/src/client.rs#L62

Reply:
Oh interesting. It seems this sends a SIGKILL which is kind of aggressive, and could cause corruption. It's occurring to me that this may be why rust-analyzer has to reindex everything every time it runs. It also looks like there isn't a way to send a SIGTERM through the built-in tokio API. It looks like [we'd have to add platform-specific code to handle this properly](https://stackoverflow.com/questions/49210815/how-do-i-send-a-signal-to-a-child-subprocess). This seems like an orthogonal issue though, so maybe I can just file another issue.